testing/test_config.py

```diff
@@ -341,6 +341,31 @@ def test_invalid_options_show_extra_information(testdir):
         "*  rootdir: %s*" % testdir.tmpdir,
     ])
 
+
+@pytest.mark.parametrize('args', [
+    ['dir1', 'dir2', '-v'],
+    ['dir1', '-v', 'dir2'],
+    ['dir2', '-v', 'dir1'],
+    ['-v', 'dir2', 'dir1'],
+])
+def test_consider_args_after_options_for_rootdir_and_inifile(testdir, args):
+    """
+    Consider all arguments in the command-line for rootdir and inifile
+    discovery, even if they happen to occur after an option. #949
+    """
+    # replace "dir1" and "dir2" from "args" into their real directory
+    root = testdir.tmpdir.mkdir('myroot')
+    d1 = root.mkdir('dir1')
+    d2 = root.mkdir('dir2')
+    for i, arg in enumerate(args):
+        if arg == 'dir1':
+            args[i] = d1
+        elif arg == 'dir2':
+            args[i] = d2
+    result = testdir.runpytest(*args)
+    result.stdout.fnmatch_lines(['*rootdir: *myroot, inifile: '])
```

Comment:
yep, thanks makes sense then
ok, then im fine with merging, should i do it or Ronny?

On Wed, Aug 26, 2015 at 5:42 PM Bruno Oliveira <notifications@github.com>
wrote:

> In testing/test_config.py
> <https://github.com/pytest-dev/pytest/pull/961#discussion_r37997786>:
>
> > +def test_consider_args_after_options_for_rootdir_and_inifile(testdir, args):
> > +    """
> > +    Consider all arguments in the command-line for rootdir and inifile
> > +    discovery, even if they happen to occur after an option. #949
> > +    """
> > +    # replace "dir1" and "dir2" from "args" into their real directory
> > +    root = testdir.tmpdir.mkdir('myroot')
> > +    d1 = root.mkdir('dir1')
> > +    d2 = root.mkdir('dir2')
> > +    for i, arg in enumerate(args):
> > +        if arg == 'dir1':
> > +            args[i] = d1
> > +        elif arg == 'dir2':
> > +            args[i] = d2
> > +    result = testdir.runpytest(*args)
> > +    result.stdout.fnmatch_lines(['*rootdir: *myroot, inifile: '])
>
> Hmmm I see.
>
> But the point of the test was to make sure if you inserted other options
> in the middle of the paths, the paths to the right should still be
> considered to determine the rootdir. What you are asking seems to be to
> test the normal semantics for rootdir, which is already tested down the
> file (in the TestRootdir class). Or am I missing something?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pytest-dev/pytest/pull/961/files#r37997786>.
>